src/components/TodoList.vue

```diff
@@ -0,0 +1,83 @@
+<template>
+  <div>
+   <v-btn @click="onShowForm" color="blue">Add Todo</v-btn>
+   <v-data-table :headers="headers" :items="todos" class="elevation-1">
+      <template v-slot:body="{ items }">
+        <tr v-for="(item, index) in items" :key="index">
+          <td>{{ item.task }}</td>
+          <td>{{ item.description }}</td>
+          <td>{{ item.assignedTo }}</td>
+          <td>
+            <v-btn color="blue" small @click="editTask(item, index)">
+              Edit
+            </v-btn>
+          </td>
+          <td>
+            <v-spacer></v-spacer>
+            <v-btn color="#FF1744" small @click="deleteTask(item)">
+              Delete
+            </v-btn>
+          </td>
+        </tr>
+      </template>
+    </v-data-table>
+      <the-form ref="test" @submit-form="submittedForm" @submit-edit="editedForm" v-if="showForm" @closed="showForm = false"> </the-form>
+  </div>
+</template>
+<script>
+import TheForm from './TheForm.vue'
+export default {
+  components: {
+    TheForm
+  },
+  name: 'TodoList',
+  data () {
+    return {
+      todos: [],
+      headers: [
+        { text: 'Task', align: 'start', value: 'task' },
+        { text: 'Description', value: 'description' },
+        { text: 'Assigned-To', value: 'assignedTo' },
+        { text: 'Edit', value: 'edit', sortable: false },
+        { text: 'Delete', value: 'delete', sortable: false }
+      ],
+      showForm: false,
+      dialog: false,
+      todoIndex: '',
+      editIndex: '',
+      edits: {
+        task: '',
+        description: '',
+        assignedTo: ''
+      }
+    }
+  },
+  methods: {
+    submittedForm (taskInput) {
+      this.todos.push(taskInput)
+    },
+    editedForm (taskInput) {
+      this.todos.splice(this.editIndex, 1, {
+        task: this.item.task,
+        description: this.item.description,
+        assignedTo: this.item.assignedTo
```

Comment:
Your taskInput contains all the edited details, but you are updating it in a wrong way, why here task: this.item.task??
It should be task: taskInput.task.
similarly for all of it you have to do the same.